### PR TITLE
update perf tests to use curried functions

### PR DESCRIPTION
**Pull request by davidchase**
update perf tests to use functions and also use es6 `const` and `let`

however a few small observations/questions... 

- merge for me just runs and doesnt display an really results other than
```sh
> node ./merge.js

merge 100000 x 10 streams
-------------------------------------------------------
```

- it seems that `switch` isnt part of the run stack? maybe we should add it but `xstream` bug would need to be fixed (seems to error)

- should `flatMap` be now `chain` for file name and contents since no more aliases ?

once we decide on those we could update the stats [README](https://github.com/mostjs/core/blob/master/test/perf/README.md)

### CODE REVIEW

**Comment by briancavalier** [comment]
> merge for me just runs and doesnt display an really results other than

I'll pull the branch and take a look.

> should flatMap be now chain for file name and contents since no more aliases ?

:+1: Yes, please!

> it seems that switch isnt part of the run stack? maybe we should add it but xstream bug would need to be fixed (seems to error)

It seems to happen in the main cujojs/most repo as well.  Maybe @staltz can help us understand if we're doing something wrong.  Here's the error I get in cujojs/most:

```
switch 10000 x 1000 streams
-------------------------------------------------------
most               1431.50 op/s ±  1.10%   (80 samples)
rx 5                  1.79 op/s ±  7.74%   (13 samples)
rx 4                 95.95 op/s ±  1.30%   (83 samples)
/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:695
        (this.inner = s)._add(this.il = new MapFlattenInner(u, this));
                         ^

TypeError: (intermediate value)(intermediate value)._add is not a function
    at MapFlattenOperator._n (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:695:26)
    at Stream._n (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:918:18)
    at FromArrayProducer._start (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:217:17)
    at Stream._add (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:988:19)
    at MapFlattenOperator._start (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:662:24)
    at Stream._add (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:988:19)
    at FoldOperator._start (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:554:18)
    at MemoryStream._add (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:1858:19)
    at LastOperator._start (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:598:18)
    at Stream._add (/Users/brian/Projects/cujojs/most/test/perf/node_modules/xstream/core.js:988:19)
```

In the meantime, we could add it, and just comment out the xstream suite, adding a note about the error.

File: test/perf/merge.js [hunk 1]
```diff
@@ -44,35 +44,35 @@ var options = {
 
 suite
   .add('most', function(deferred) {
-    var streams = a.map(most.from);
-    runners.runMost(deferred, most.mergeArray(streams).reduce(sum, 0));
+    const streams = map(from, a);
```
Comment (briancavalier) [inline test/perf/merge.js]: `a` is an Array here, so this line should be: `const streams = a.map(from)`.  That's why the test is bailing out immediately--not sure why the failure is silent, tho :/

Comment (davidchase) [inline test/perf/merge.js]: doh! thanks for catching that one

Comment (briancavalier) [inline test/perf/merge.js]: I had to stare at it for a bit!  Pretty hard to see it after you've gotten accustomed to using function APIs everywhere :)

**Comment by davidchase** [comment]
@briancavalier found the issue with [switch](https://github.com/mostjs/core/blob/277b59d1ca09488179b26fc6128621eb012cd06b/test/perf/switch.js#L62) it uses the `bacon.fromArray` instead of xstream version... check the `cujojs/most`

**Comment by briancavalier** [comment]
> it uses the bacon.fromArray instead of xstream version... check the cujojs/most

Oh jeez 🤦‍♂️ sorry about that.

**Comment by davidchase** [comment]
no worries should have caught it earlier... had to stare at it as well 😄

**Comment by davidchase** [comment]
Below is my results for running the whole thing

<details>
<summary>Updated Perfs</summary>

```sh
λ npm start

> most-perf@0.10.0 start /Users/davidchase/Desktop/mostjs/core/test/perf
> npm run filter-map-reduce && npm run chain && npm run concatMap && npm run merge && npm run merge-nested && npm run zip && npm run scan && npm run slice && npm run skipRepeats && npm run switch


> most-perf@0.10.0 filter-map-reduce /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./filter-map-reduce

filter -> map -> reduce 1000000 integers
-------------------------------------------------------
most                484.22 op/s ±  1.50%   (77 samples)
rx 4                  1.30 op/s ±  4.40%   (11 samples)
rx 5                 10.00 op/s ±  1.54%   (49 samples)
xstream              15.80 op/s ±  1.88%   (73 samples)
kefir                 9.27 op/s ±  2.51%   (47 samples)
bacon                 0.79 op/s ±  1.41%    (8 samples)
highland              5.01 op/s ±  2.01%   (28 samples)
-------------------------------------------------------

> most-perf@0.10.0 chain /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./chain.js

chain 1000 x 1000 streams
-------------------------------------------------------
most                140.98 op/s ±  1.42%   (78 samples)
rx 4                  0.67 op/s ±  5.21%    (8 samples)
rx 5                 25.26 op/s ±  1.08%   (60 samples)
xstream              10.98 op/s ±  1.76%   (53 samples)
kefir                 8.93 op/s ±  0.73%   (45 samples)
bacon                 0.67 op/s ±  2.07%    (8 samples)
highland              0.09 op/s ±  5.28%    (5 samples)
-------------------------------------------------------

> most-perf@0.10.0 concatMap /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./concatMap.js

concatMap 1000 x 1000 streams
-------------------------------------------------------
most                103.63 op/s ±  2.12%   (77 samples)
rx 4                  1.26 op/s ±  2.96%   (11 samples)
rx 5                 26.81 op/s ±  1.43%   (63 samples)
xstream              11.06 op/s ±  0.93%   (54 samples)
kefir                 9.37 op/s ±  1.80%   (47 samples)
bacon                 0.70 op/s ±  2.34%    (8 samples)
-------------------------------------------------------

> most-perf@0.10.0 merge /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./merge.js

merge 100000 x 10 streams
-------------------------------------------------------
most                279.31 op/s ±  1.45%   (77 samples)
rx 4                  1.13 op/s ±  2.88%   (10 samples)
rx 5                 28.33 op/s ±  0.98%   (66 samples)
xstream              16.18 op/s ±  1.38%   (73 samples)
kefir                10.29 op/s ±  1.85%   (51 samples)
bacon                 0.66 op/s ±  2.48%    (8 samples)
-------------------------------------------------------

> most-perf@0.10.0 merge-nested /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./merge-nested.js

merge nested streams w/depth 2, 5, 10, 100 (10000 items in each stream)
-------------------------------------------------------
most (depth 2)     9533.71 op/s ±  1.42%   (77 samples)
most (depth 5)     4877.75 op/s ±  1.28%   (78 samples)
most (depth 10)    2644.43 op/s ±  1.13%   (76 samples)
most (depth 100)    279.08 op/s ±  1.86%   (78 samples)
rx 4 (depth 2)       42.18 op/s ±  1.76%   (65 samples)
rx 4 (depth 5)       16.07 op/s ±  0.99%   (72 samples)
rx 4 (depth 10)       6.68 op/s ±  2.44%   (35 samples)
rx 4 (depth 100)      0.16 op/s ±  2.85%    (5 samples)
rx 5 (depth 2)      588.81 op/s ±  2.45%   (77 samples)
rx 5 (depth 5)      237.04 op/s ±  1.95%   (76 samples)
rx 5 (depth 10)     103.51 op/s ±  1.36%   (77 samples)
rx 5 (depth 100)      1.73 op/s ±  3.25%   (13 samples)
xstream (depth 2)   472.85 op/s ±  2.32%   (80 samples)
xstream (depth 5)   188.94 op/s ±  1.82%   (77 samples)
xstream (depth 10)   79.05 op/s ±  1.85%   (72 samples)
xstream (depth 100)    1.42 op/s ±  2.76%   (11 samples)
kefir (depth 2)     288.50 op/s ±  1.90%   (80 samples)
kefir (depth 5)      97.72 op/s ±  1.90%   (73 samples)
kefir (depth 10)     35.07 op/s ±  2.20%   (78 samples)
kefir (depth 100)     0.46 op/s ±  3.46%    (7 samples)
bacon (depth 2)      19.36 op/s ±  3.08%   (49 samples)
bacon (depth 5)       8.32 op/s ±  2.02%   (42 samples)
bacon (depth 10)      3.68 op/s ±  1.26%   (22 samples)
bacon (depth 100)     0.08 op/s ±  3.30%    (5 samples)
highland (depth 2)    1.98 op/s ± 10.73%   (14 samples)
highland (depth 5)    1.96 op/s ± 12.11%   (14 samples)
highland (depth 10)    0.16 op/s ±  3.53%    (5 samples)
-------------------------------------------------------

> most-perf@0.10.0 zip /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./zip.js

zip 2 x 100000 integers
-------------------------------------------------------
most                104.83 op/s ±  2.75%   (77 samples)
rx 4                  2.10 op/s ±  2.00%   (15 samples)
rx 5                  0.21 op/s ±  1.20%    (6 samples)
kefir                 0.45 op/s ± 48.21%    (7 samples)
bacon                 0.10 op/s ±  4.37%    (5 samples)
highland              0.06 op/s ± 11.54%    (5 samples)
-------------------------------------------------------

> most-perf@0.10.0 scan /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./scan.js

scan -> reduce 1000000 integers
-------------------------------------------------------
most                287.88 op/s ±  1.08%   (79 samples)
rx 4                  1.35 op/s ±  1.66%   (11 samples)
rx 5                 11.14 op/s ±  1.09%   (54 samples)
xstream               8.63 op/s ±  1.30%   (44 samples)
kefir                14.48 op/s ±  1.17%   (66 samples)
bacon                 0.63 op/s ±  2.21%    (8 samples)
highland              5.12 op/s ±  3.62%   (29 samples)
-------------------------------------------------------

> most-perf@0.10.0 slice /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./slice.js

skip(n/4) -> take(n/2) 1000000 integers
-------------------------------------------------------
most                108.64 op/s ±  1.03%   (70 samples)
rx 4                  2.03 op/s ±  0.89%   (14 samples)
rx 5                 43.00 op/s ±  1.39%   (66 samples)
xstream              18.44 op/s ±  2.38%   (43 samples)
kefir                10.55 op/s ±  7.94%   (53 samples)
bacon                 0.80 op/s ±  4.21%    (8 samples)
highland              6.30 op/s ±  2.23%   (34 samples)
-------------------------------------------------------

> most-perf@0.10.0 skipRepeats /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./skipRepeats.js

skipRepeats -> reduce 2 x 1000000 integers
-------------------------------------------------------
most                313.07 op/s ±  1.04%   (76 samples)
rx 4                  1.65 op/s ±  1.72%   (13 samples)
rx 5                 11.09 op/s ±  1.09%   (54 samples)
xstream              20.24 op/s ±  1.57%   (50 samples)
kefir                11.63 op/s ±  1.25%   (56 samples)
bacon                 0.82 op/s ±  1.99%    (9 samples)
-------------------------------------------------------

> most-perf@0.10.0 switch /Users/davidchase/Desktop/mostjs/core/test/perf
> node ./switch.js

switch 10000 x 1000 streams
-------------------------------------------------------
most               1694.91 op/s ±  5.09%   (81 samples)
rx 5                  1.62 op/s ±  8.60%   (12 samples)
rx 4                 89.70 op/s ±  1.75%   (79 samples)
xstream               1.21 op/s ±  5.38%   (10 samples)
kefir                 0.96 op/s ±  2.82%    (9 samples)
bacon                 0.02 op/s ±  4.49%    (5 samples)
-------------------------------------------------------
```

</details>

**Comment by briancavalier** [comment]
Cool.  Those results look on par with what I see on my machine, too.

**Pull request merged**